src/network/networkd-dhcp4.c

```diff
@@ -37,6 +37,9 @@ void network_adjust_dhcp4(Network *network) {
         if (network->dhcp_use_gateway < 0)
                 network->dhcp_use_gateway = network->dhcp_use_routes;
 
+        if (network->dhcp_use_dnr < 0)
+                network->dhcp_use_dnr = network->dhcp_use_dns;
```

Comment:
so i'd prefer that there was a static inline helper that resolves this, and that we'd only check the field through that, i.e.:

```c
static inline network_dhcp_use_dnr(Network *n) {
        assert(n);
        return network->dhcp_use_dnr < 0 ? network->dhcp_use_dns : network->dhcp_use_dnr;
```

i undestand you copied that logic from the use_gateway/use_routes lines above, but i think that should be adjust the same way.

i.e. i'd always store the whole info in the structures and resolve unspecified stuff through a helper at access rather than already adjust the structures to resolve things. (i mean, i think it's fine suppressing entirely redundant info, but i think this here isn't precisely redundant)